docs/setup-teardown.md

```diff
@@ -0,0 +1,120 @@
+# Setup and Teardown
+
+The Playwright test runner provides two mechanisms for setup and teardown.
+
+- [Fixtures](#fixtures): Fixtures run setup/teardown code for test functions and are accessed as arguments by the test function.
+  - [Example: Test data](#example-load-test-data-in-memory)
+  - [Example: Authentication](#example-authentication)
+- [Classic BDD](#classic-bdd): Use `beforeEach`, `beforeAll` and their after- equivalents.
+
+Repeat a step for all tests (beforeEach/afterEach, test fixtures)
+One-time setup for a test file (beforeAll/afterAll)
+One-time setup for a worker (worker fixtures)
+
+## Fixtures
+
+This the recommended approach for setup and teardown in the Playwright test runner. Fixtures have a few benefits over classic BDD setup, especially for end-to-end testing.
+
+- Fixtures minimize configuration overload: By centralizing test configuration, fixtures ensure your test functions are minimal and ease maintenance.
+- Fixtures promote best-practices: Fixtures promote test-level isolation for pages used by end-to-end tests. This ensures failures in one test do not cascade.
+- Fixtures are modular and flexible.
+
+Fixtures can have different scope levels, depending on how they fit into the testing lifecycle.
+
+- Test-level fixtures: These fixtures are created for every test function that uses them.
+  - `context` and `page` are built-in test-level fixtures. See all [built-in fixtures][default-args].
+- Worker-level fixtures: A worker process is a unit of parallelization and runs a set of tests. Worker-level fixtures are run once per worker.
```

Comment:
I prefer to look at worker fixtures as an optimization, where you can only run it once and share some heavyweight resource between tests.

Reply:
Thanks! I broadly agree. I could position it as an optimization here.

Looking at user scenarios, I believe we need to highlight test-level fixtures and global setup (filed as https://github.com/microsoft/playwright-test/issues/177), and not worker-level fixtures.